chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb

```diff
@@ -49,17 +49,26 @@ touch $RUNFILE
 trap cleanup INT EXIT
 
 <% if @use_ha %>
-# Shutdown pacemaker so the remaining OpenStack services are stopped
+# Shutdown corosync so the remaining OpenStack services are stopped
 # This is needed so that the zypper dup won't trigger any db migrations on package update
-service pacemaker stop
+service corosync stop
 
 ret=$?
 if [ $ret != 0 ] ; then
-    log "Error occured during shutting down pacemaker service"
+    log "Error occured during shutting down corosync service"
     echo $ret > $UPGRADEDIR/crowbar-pre-upgrade-failed
     exit $ret
 fi
 
+# Make sure sbd is shut down (if it was configured) before proceeding with next actions
+if systemctl is-enabled sbd ; then
+    SBD_PIDFILE=$(sed -n 's/PIDFile=//p' /usr/lib/systemd/system/sbd.service)
```

Comment:
It's a little safer to use use ```s/^PIDFile=//p``` here I think.

And it theory it is possible to override the settings here by configuration in ```/etc/systemd/....```. How about using ```systemctl show sbd.service --property PIDFile | sed -n 's/^PIDFile=//p``` or maybe even ```eval $(systemctl show sbd.service --property PIDFile)```